indico/web/client/js/react/components/PersonDetailsModal.jsx

```diff
@@ -170,11 +181,114 @@ PersonDetailsModal.propTypes = {
   onSubmit: PropTypes.func.isRequired,
   onClose: PropTypes.func.isRequired,
   person: PropTypes.object,
+  eventId: PropTypes.number,
+  otherPersons: PropTypes.array,
   hasPredefinedAffiliations: PropTypes.bool.isRequired,
   hideEmailField: PropTypes.bool,
 };
 
 PersonDetailsModal.defaultProps = {
   person: undefined,
+  eventId: null,
+  otherPersons: [],
   hideEmailField: false,
 };
+
+function EmailField({shouldValidate, validateUrl, currentEmail, otherPersons}) {
+  let response, msg;
+  const [message, setMessage] = useState({status: '', message: ''});
+
+  const validateEmail = useDebouncedAsyncValidate(async email => {
+    email = email.trim();
+
+    if (email === '' || email === currentEmail) {
+      setMessage({status: '', message: ''});
+      return;
+    }
+
+    // Prevent adding the same person twice
+    if (otherPersons.some(p => p.email === email)) {
+      msg = Translate.string('There is already a person with this email in the list.');
+      setMessage({
+        status: 'error',
+        message: msg,
+      });
+      return msg;
+    }
+
+    setMessage({
+      status: '',
+      message: Translate.string('Checking email address...'),
+    });
+
+    try {
+      response = await indicoAxios.get(validateUrl, {
+        params: {email},
+      });
+    } catch (error) {
+      return handleAxiosError(error);
+    }
+
+    const {data} = response;
+    const {status, conflict} = data;
+    if (conflict === 'person-already-exists') {
+      msg = (
+        <Translate>
+          This entry will be associated with the user{' '}
+          <Param name="name" wrapper={<strong />} value={data.person_name} />.
+        </Translate>
+      );
+    } else if (conflict === 'user-already-exists') {
+      msg = Translate.string(
+        'An Indico user with this email already exists. You can alternatively add them using the search option.'
```

Comment:
Could we add a shortcut here that you can search for the user by email and select them with a single click? :)